pkg/webui/console/store/actions/device.js

```diff
@@ -0,0 +1,29 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+export const GET_DEV = 'GET_DEVICE'
+export const GET_DEV_SUCCESS = 'GET_DEVICE_SUCCESS'
+export const GET_DEV_FAILURE = 'GET_DEVICE_FAILURE'
+
+export const getDevice = (application_id, device_id, selector, options) => (
```

Comment:
```suggestion
export const getDevice = (appId, deviceId, selector, options) => (
```

Reply:
```suggestion
export const getDevice = (appId, deviceId, meta: { selector, options }) => (
```
There is absolutely no reason to pollute the action with implementation specific entires

Reply:
I would say the selector is a first-class parameter in get actions. Everything else can be put into `options`. It is then reflecting the signature of the sdk service method. If we should replace our api tree with the actual sdk instance, this will make things a lot easier.